### PR TITLE
Added Cantor keyboard

### PR DESCRIPTION
**Pull request by diepala**
Added support for the Cantor keyboard: https://github.com/diepala/cantor

## Description

Added a new folder inside keyboards, named cantor, with the necessary files to build the firmware for the Cantor keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/cantor/info.json [hunk 1]
```diff
@@ -0,0 +1,70 @@
+{
+    "manufacturer": "Diego Palacios",
+    "keyboard_name": "cantor",
+    "maintainer": "diepala",
+    "bootloader": "stm32-dfu",
+    "diode_direction": "COL2ROW",
+    "features": {
+        "bootmagic_lite": true,
```
Comment (zvecr) [inline keyboards/cantor/info.json:8]:
```suggestion
        "bootmagic": true,
```

File: keyboards/cantor/info.json [hunk 2]
```diff
@@ -0,0 +1,70 @@
+{
+    "manufacturer": "Diego Palacios",
+    "keyboard_name": "cantor",
+    "maintainer": "diepala",
+    "bootloader": "stm32-dfu",
+    "diode_direction": "COL2ROW",
```
Comment (zvecr) [inline keyboards/cantor/info.json:6]:
```suggestion
```

File: keyboards/cantor/rules.mk [hunk 3]
```diff
@@ -0,0 +1,4 @@
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+
```
Comment (zvecr) [inline keyboards/cantor/rules.mk:2]:
```suggestion
```

File: keyboards/cantor/mcuconf.h [hunk 4]
```diff
@@ -0,0 +1,31 @@
+
```
Comment (zvecr) [inline keyboards/cantor/mcuconf.h:1]:
```suggestion
```

File: keyboards/cantor/cantor.h [hunk 5]
```diff
@@ -0,0 +1,35 @@
+// Copyright 2022 Diego Palacios (@diepala)
```
Comment (zvecr) [inline keyboards/cantor/cantor.h:1]: This file should be removed as you have matrix data in `info.json`

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/cantor/cantor.h (hunk 5 above)
Comment (diepala) [reply keyboards/cantor/cantor.h:1]: Removing this file produces errors

Comment (drashna) [reply keyboards/cantor/cantor.h:1]: Remove the layout macrco specifically, then

Comment (diepala) [reply keyboards/cantor/cantor.h:1]: It won't copmile:
```
☒ Matrix data out of bounds for layout LAYOUT_split_3x6_3 at index 6 (k4A): 4, 0
gmake[1]: *** [builddefs/build_keyboard.mk:335: .build/obj_cantor/src/layouts.h] Error 1
```

**Comment by vogler** [comment]
Just to report interest in this getting merged:
I wanted to make a hand-wired split keyboard based on two STM32 black pill boards.
I searched for STM32 in QMK Toolbox and flashed some, but they did not work.
(https://qmk.fm/keyboards/ is not great for searching - is there some better index?)
This PR works (but trying it out required installing all of qmk and its deps for building).

**Comment by jcbird** [comment]
Commenting to express my interest in getting this PR merged as well. Lots of good fixes for blackpill based boards. Thanks!

**Review by zvecr — APPROVED** [review]

File: keyboards/cantor/readme.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
+# cantor
+
+![cantor](https://i.imgur.com/Uvxm3zV.jpg)
```
Comment (fauxpark) [inline keyboards/cantor/readme.md:3]:
```suggestion
![cantor](https://i.imgur.com/Uvxm3zVh.jpg)
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/cantor/info.json [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+        "nkro": true
+    },
+    "processor": "STM32F401",
+    "url": "https://github.com/diepala/cantor",
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x0000",
+        "vid": "0xFEED"
+    },
+    "layouts": {
```
Comment (zvecr) [inline keyboards/cantor/info.json:21]:
```suggestion
    "community_layouts": ["split_3x6_3"],
    "layouts": {
```

**Review by drashna — APPROVED** [review]

**Pull request merged**